src/StepFunctionGraph/Digraph.ts

```diff
@@ -0,0 +1,119 @@
+interface Vertex {
+    vertex: number;
+    data: any;
+}
+
+interface OutDegree {
+    count: number;
+    outVertices: number[];
+}
+
+class Digraph {
+    V: number;
+    E: number;
+    adjacencyMatrix: number[][];
+    vertexDataMap: Map<number, any>;
+    indegree: number[][];
+
+    constructor(vertices?: number) {
+        if (vertices && vertices < 0) throw new Error('Number of vertices in a Digraph must be non-negative');
+
+        // Vertex and Edge counters
+        this.V = 0;
+        this.E = 0;
+
+        // This is the heart of the Digraph, storing outdegree vertices at an index which is the vertex that points to it
+        this.adjacencyMatrix = [];
+        this.vertexDataMap = new Map<number, any>();
+
+        // Indegree represents the vertices that point to a vertex where the index is vertex we want values for
```

Comment:
super nitpick. I think "Where the index is vertex" should have a 'the' to read "Where the index is the vertex"

Reply:
Thanks!